server/eclipse-project/src/main/java/us/freeandfair/corla/Main.java

```diff
@@ -254,16 +274,62 @@ private void activateEndpoints() {
   }
   
   /**
-   * Initializes the ASM. If it cannot be read from the database, it is created from 
-   * scratch and committed to the database.
+   * Restores an ASM's state or persists it in the database.
    * 
-   * @exception IllegalStateException if we can't get the ASM. 
+   * @param the_asm The ASM.
+   * @param the_state The persistent state to restore, or null to persist
+   *  the state to the database.
+   * @exception PersistenceException if the state cannot be persisted.
    */
-  private void initializeASM() throws IllegalStateException {
+  private void restoreOrPersistState(final AbstractStateMachine the_asm,
```

Comment:
I'm a little confused by the idea behind this method. It does opposite things, depending on if `the_state` is `null` or not, right? I guess it is private so it doesn't really matter.